client/src/cmdlfem4x70.c

```diff
@@ -29,7 +29,65 @@
 #define LOCKBIT_0 BITMASK(6)
 #define LOCKBIT_1 BITMASK(7)
 
-#define INDEX_TO_BLOCK(x) (((32-x)/2)-1)
+#define BYTE_ARRAY_INDEX_TO_BLOCK(x) ((31-(x))/2)
+
+// result from CMD_LF_EM4X70_INFO
+typedef struct _em4x70_tag_info {
+    /// <summary>
+    /// The full data on an em4x70 the tag.
+    /// [31] == Block 15 MSB == UM2₆₃..UM2₅₆
+    /// [30] == Block 15 LSB == UM2₅₅..UM2₄₈
+    /// [29] == Block 14 MSB == UM2₄₇..UM2₄₀
+    /// [28] == Block 14 LSB == UM2₃₉..UM2₃₂
+    /// [27] == Block 13 MSB == UM2₃₁..UM2₂₄
+    /// [26] == Block 13 LSB == UM2₂₃..UM2₁₆
+    /// [25] == Block 12 MSB == UM2₁₅..UM2₀₈
+    /// [24] == Block 12 LSB == UM2₀₇..UM2₀₀
+    /// [23] == Block 11 MSB == Pin₃₁..Pin₂₄
+    /// [22] == Block 11 LSB == Pin₂₃..Pin₁₆
+    /// [21] == Block 10 MSB == Pin₁₅..Pin₀₈
+    /// [20] == Block 10 LSB == Pin₀₇..Pin₀₀
+    /// [19] == Block  9 MSB == Key₉₅..Key₈₈
+    /// [18] == Block  9 LSB == Key₈₇..Key₈₀
+    /// [17] == Block  8 MSB == Key₇₉..Key₇₂
+    /// [16] == Block  8 LSB == Key₇₁..Key₆₄
+    /// [15] == Block  7 MSB == Key₆₃..Key₅₆
+    /// [14] == Block  7 LSB == Key₅₅..Key₄₈
+    /// [13] == Block  6 MSB == Key₄₇..Key₄₀
+    /// [12] == Block  6 LSB == Key₃₉..Key₃₂
+    /// [11] == Block  5 MSB == Key₃₁..Key₂₄
+    /// [10] == Block  5 LSB == Key₂₃..Key₁₆
+    /// [ 9] == Block  4 MSB == Key₁₅..Key₀₈
+    /// [ 8] == Block  4 LSB == Key₀₇..Key₀₀
+    /// [ 7] == Block  3 MSB == ID₃₁..ID₂₄
+    /// [ 6] == Block  3 LSB == ID₂₃..ID₁₆
+    /// [ 5] == Block  2 MSB == ID₁₅..ID₀₈
+    /// [ 4] == Block  2 LSB == ID₀₇..ID₀₀
+    /// [ 3] == Block  1 MSB == L₁ L₀ UM1₂₉..UM1₂₄
+    /// [ 2] == Block  1 LSB == UM1₂₃..UM1₁₆
+    /// [ 1] == Block  0 MSB == UM1₁₅..UM1₀₈
+    /// [ 0] == Block  0 LSB == UM1₀₇..UM1₀₀
+    /// </summary>
+    /// <remarks>
+    /// When moving to C++, strongly consider adding
+    /// const helper functions to extract a given
+    /// block of data into native uint16_t.
+    /// See also the print_info_result() function
+    /// for visual presentation of the data.
+    /// </remarks>
+    uint8_t Raw[32];
+} em4x70_tag_info;
+
+// result from CMD_LF_EM4X70_AUTH is defined in ID48LIB (id48.h) as ID48LIB_GRN 
+
+typedef struct _em4x70_cmd_helper_options {
+    // variables named based on default value (zero-init) being false
+    uint32_t use_parity : 1;             // default is false ... no parity
+    uint32_t disable_error_output : 1;   // default is false ... errors/warnings are output using PrintAndLogEx()
+    uint32_t disable_all_output   : 1;   // default is false ... success often his output using PrintAndLogEx()
+    uint32_t : 29; // padding
```

Comment:
it almost looks like you are pushing your own printing logic..  Joining a third one to our DEBUG, and VERBOSE patterns...   Please don't.

Reply:
This is not the intention.  Rather, these helper functions are going to be called from multiple points.   Some of these would expect errors to occur, and some do not.  This is essentially a flag saying, "For this caller, never dump errors to the log" ... which makes sense when the code is expecting the operation will fail most of the time.  There is no reason to clutter the log.   This is a different purpose than DEBUG or VERBOSE ... this is a code path decision.

The simple example is for code that will wait for the presence of a tag.   Why fill the log with thousands of useless TIMEOUT messages?  Of course, such a simple function can be easily duplicated with different printing options.  But then more complex code paths, such as, "Wait for a tag, check if it's writable, then write block X, then auth, then write Block X, then ..." is much more complex, and it can be helpful to avoid duplicating all the helper functions.

I didn't really like the `opts` argument, so I'm open to ideas.   I can generate `function_x_no_output()` and `function_x_only_error_output()` variants of functions instead?   But, code duplication didn't seem to be the right way either?